pipeliner.go

```diff
@@ -194,15 +192,32 @@ func (p *pipeliner) flush(reqs []CmdAction) []CmdAction {
 
 type pipelinerCmd struct {
 	CmdAction
-	resCh chan error
+	res    error
+	resCh  chan error
+	resSet bool
+}
+
+func (p *pipelinerCmd) setRes(err error) {
+	if p.resSet {
+		return
+	}
+	p.res = err
+	p.resSet = true
+}
+
+func (p *pipelinerCmd) send() {
```

Comment:
Just for clarity I think `sendRes` might be a better name. Also, are there any cases where `resSet == true` and `p.res == nil`? I can't think how that would happen with the given logic, but maybe I'm missing something. If not, maybe it'd be simpler to drop `resSet` and just check that `res == nil` before writing it.

Reply:
`p.res` will be `nil` whenever the was no error, which should be most cases. If we were to drop `resSet` we would need to keep track where the `pipelinerPipine` exits with an error, otherwise we may set an error on a `pipelinerCmd` that had no error before.

Maybe another name than `resSet` might be better.